lib/debootstrap.sh

```diff
@@ -590,16 +590,33 @@ create_image()
 	cp $SDCARD/etc/armbian.txt $DESTIMG
 	mv ${SDCARD}.raw $DESTIMG/${version}.img
 
-	if [[ $COMPRESS_OUTPUTIMAGE == yes && $BUILD_ALL != yes ]]; then
-		[[ -f $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME ]] && cp $DEST/images/$CRYPTROOT_SSH_UNLOCK_KEY_NAME $DESTIMG/
-		# compress image
+	if [[ $COMPRESS_OUTPUTIMAGE == yes ]]; then
+		COMPRESS_OUTPUTIMAGE="sha,7z"
+	fi
+
+	if [[ $COMPRESS_OUTPUTIMAGE == *sha* && $BUILD_ALL != yes ]]; then
 		cd $DESTIMG
+		display_alert "SHS256 calculating" "${version}.img" "info"
 		sha256sum -b ${version}.img > sha256sum.sha
+		cp sha256sum.sha "$DEST/images/${version}.img.sha"
 		if [[ -n $GPG_PASS ]]; then
-			echo $GPG_PASS | gpg --passphrase-fd 0 --armor --detach-sign --pinentry-mode loopback --batch --yes ${version}.img
+			echo $GPG_PASS | gpg --passphrase-fd 0 --armor --detach-sign --pinentry-mode loopback --batch --yes ${version}.img || exit 1
 		fi
-			display_alert "Compressing" "$DEST/images/${version}.img" "info"
-		7za a -t7z -bd -m0=lzma2 -mx=3 -mfb=64 -md=32m -ms=on $DEST/images/${version}.7z ${version}.key ${version}.img armbian.txt *.asc sha256sum.sha >/dev/null 2>&1
+		cd ..
```

Comment:
`cd ..` is hard to follow if this code gets modified in the future.. Isn't there a better way to describe it? If I got it right we should be in .tmp after cd .. right?

Reply:
I can use pushd/popd as I use it in my scripts usually.
```cd ..``` has origin from previous version of this script.